### PR TITLE
Pull actively info-applying functions out of rz_core_bin_info()

### PR DESCRIPTION
**Pull request by thestr4ng3r**

### CODE REVIEW

File: librz/core/cbin.c [hunk 1]
```diff
@@ -321,33 +345,90 @@ RZ_API int rz_core_bin_set_by_name(RzCore *core, const char *name) {
...
+	rz_asm_use(r->rasm, arch);
+
+	// ----
+	// inlined: rz_core_bin_info(r, RZ_CORE_BIN_ACC_ALL, NULL, RZ_MODE_SET, va, NULL, NULL);
+	ut64 loadaddr = binobj ? binobj->loadaddr : UT64_MAX;
+
+	// use our internal values for va
+	va = va ? VA_TRUE : VA_FALSE;
+
+	rz_core_bin_apply_strings(r, binfile);
```
Comment (ret2libc) [inline librz/core/cbin.c:379]: why all these calls?

Comment (thestr4ng3r) [reply librz/core/cbin.c:379]: This is all the code that `rz_core_bin_info(r, RZ_CORE_BIN_ACC_ALL, NULL, RZ_MODE_SET, va, NULL, NULL)`, which was there before was calling. The plan is to make it into individual functions `rz_core_bin_apply_*()` per info type and only for applying info, while `rz_core_bin_info()` will only do printing.

Comment (ret2libc) [reply librz/core/cbin.c:379]: Ah, YES! If this is to split the set and the printing, then yes please :)

**Review by ret2libc — APPROVED** [review]

File: librz/core/cbin.c (hunk 1 above)
Comment (thestr4ng3r) [reply librz/core/cbin.c:379]: Exactly this.

**Review by Maijin — APPROVED** [review]

File: librz/core/cbin.c [hunk 2]
```diff
@@ -29,6 +29,30 @@
 	if (binfile && binfile->rbin && binfile->rbin->verbose) \
 	eprintf
 
+static RZ_NULLABLE RZ_BORROW const RzList *core_bin_strings(RzCore *r, RzBinFile *file);
+static void _print_strings(RzCore *r, const RzList *list, PJ *pj, int mode, int va);
+static bool bin_raw_strings(RzCore *r, PJ *pj, int mode, int va);
+static int bin_info(RzCore *r, PJ *pj, int mode, ut64 laddr);
+static int bin_main(RzCore *r, PJ *pj, int mode, int va);
+static int bin_dwarf(RzCore *core, PJ *pj, int mode);
```
Comment (Maijin) [inline librz/core/cbin.c:37]: Would it be possible to sort this list somehow?

Comment (thestr4ng3r) [reply librz/core/cbin.c:37]: Yes, but these declarations are also only temporary until the are all unfolded and refactored.

**Comment by thestr4ng3r** [comment]
minstr/maxstr/filter is missing in apply_strings, putting this to draft again until I figured out how to do that best without code duplication.

**Comment by XVilka** [comment]
It looks good. In the future we might consider to split `librz/core/cmd_*.c` files into the different directory than `librz/core/cbin.c`, `librz/core/canalysis.c`, or vice versa.

File: librz/core/cbin.c (hunk 2 above)
Comment (Maijin) [reply librz/core/cbin.c:37]: ok

**Comment by Maijin** [comment]
@thestr4ng3r if you're refactoring, may as well use the table API to have the ability to sort per column etc. 👍

**Comment by thestr4ng3r** [comment]
Table API is a good idea, but should be done seperately.

**Review by Maijin — APPROVED** [review]

**Pull request merged**